### DISCUSSION TITLE
Heat & Efficiency

### DISCUSSION THREAD

**Discussion by alexanderson1993** [Thorium-Sim/thorium-nova#369] [Ideas]
Heat is intrinsically tied to efficiency - the hotter a system gets, the more its efficiency decreases due to entropy. And Efficiency is tied to power - the lower the efficiency, the more power is required to make a system operate at normal performance. The excess power has to go somewhere, which usually means turning into heat.

In Thorium Classic, excess heat slowly dissipated, with the option to flush coolant through the system to cool it down immediately. That's not really how cooling systems work, though. They're elaborate loops of heat pipes/pumps which use a refrigerant like ammonia to transfer heat to radiators which radiate the heat out into space. 

### A simple heating system

For simplicity, we could assume that all systems that generate heat have their own dedicated radiators and coolant loops. Systems passively dissipate their heat through their own radiator system. But this leaves little room for adding interesting gameplay elements aside from just turning off systems and waiting for them to cool.

Each system would need to have a supply of coolant in order to properly dissipate heat. During the game, coolant could leak which would decrease how well the heat dissipates until coolant is restored, giving the engineer one more thing to worry about.

### Heat/Coolant Routing

A more complicated option would be a coolant routing mini game, much like power routing, but perhaps a little simpler. We could model the movement of coolant units through the ship, where each unit has a temperature that heats up in the system and cools down in radiators. There could be many radiators of different capacities that the coolant can be routed to. Each radiator can be configured to decrease the temperature of the coolant by a certain amount each tick, down to 200 K (the freezing point of ammonia).

> Most spacecraft radiators reject between 100 and 350 W of internally generated electronics waste heat per square meter. ~Wikipedia

For example, Warp engines need exceptionally large radiators for their large heat use, but other systems could use the warp engine radiators as well. 

This doesn't really have too much value to the game - ultimately the ship has X amount of heat dissipation capability, heat flows in a gradient from hot to cold, so having one massive radiator and attaching all of the systems to it would work just as well as having several smaller radiators - unless you were to take temperature tolerances into account (each system needs to be between two temperatures in order to function correctly), but that seems a bit much.

Because of that, I prefer for heat and coolant should follow the simplified model - coolant exists within its own system, heat is radiated out into space automatically, and stays isolated from other systems. Shout out in the comments if you have a different preference.

### Stealthy while hot

The interesting bit comes from trying to be stealthy. The ship can be equipped with a stealth field, which masks its presence to other ships. However, as a matter of gameplay, it shouldn't be infallible. Using projectile systems, like torpedoes and probes, are a dead giveaway. 

Heat is also a factor in stealth. If the ship is emitting heat in the form of infrared radiation, it should easily be picked up by nearby ships. To limit the probability of detection, the ship has to mask that heat somehow.

To me, the simplest solution is heat storage. Assuming you had some kind of magic coolant container that perfectly insulates hot (or cold) coolant, a clever crew could use cargo containers to transfer hot coolant _out_ of a system, replacing it with cool coolant. This would dramatically reduce the heat radiated by the ship, keeping it stealthy. The hot coolant would remain in storage at the same temperature until it is transferred back into the system to be radiated out into space. Of course, you can only do this as long as you have cold coolant to transfer heat into.

### Tying up loose ends

- Coolant should exist within its own system. As the system is used, its heat increases.
- How much heat increases is affected by efficiency.
- Efficiency decreases at a faster rate due to entropy when the system is hot.
- Coolant is an inventory item which includes a temperature property and a heat density property.
- The more coolant that is present in a system (via cargo transfer), the faster the system can cool as heat is transferred into the system's radiator. The radiator can only release so much heat into space, so at a certain point, adding more coolant does nothing to release more heat.
- One way systems can be damaged is through coolant leaks, which causes coolant to become unusable, requiring spare coolant to be transferred to the system once the leak is fixed.
- Coolant can be transferred out of systems. The heat of the coolant inventory item remains at the temperature it was at when it was removed because of space-magic insulating containers. This can be used to rapidly cool a system or keep radiators from emitting heat into space.
- This implies that the heat output of ships is something that can easily be detected through sensors.

---

I think that's all? Feedback, comments, and suggestions welcome.

**Comment by 1deejay** [comment]
Heat management probably doesn't have to be an intrusive game play thing, more of something to keep track of. For example; normal operation of the ship there should be no concern about overheating.

You can tie part of it to power consumption which is a lot of computer heat issues. Yeah, your phasers will give more of a punch, but they will need to cool down for longer. Of course, if you override the safeties you can get a big blast at risk of damaging them.

It can also be part of a system taking on damage is that it dissipates its heat less efficiently. 

The heat sink method of stealth is covered in Mass Effect as well. Your heat sinks can only handle so much heat before you cook all the people inside of the ship so it's a good idea. Plus, different ships can be configured with the ability to handle heat better like any other system.

**Reply by alexanderson1993** [reply]
I like the idea of having literal safeties in place the you have to intentionally override in order to push systems beyond limits. That's a good UX consideration to make it so crews know the consequences of what they're about to do.

> It can also be part of a system taking on damage is that it dissipates its heat less efficiently.

For the most part, damage decreases efficiency, which does impact heat generation. So it's not so much dissipating heat less, but generating heat more. I see it as a net-neutral difference, depending on which way you want to go with it.
